### PR TITLE
Maybe fix docs build cache?

### PR DESCRIPTION
**Pull request by alasdairwilson**
Obviously run.id will never work but what will...

### CODE REVIEW

**Comment by nabobalis** [comment]
Why would run.id never work?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -59,7 +71,7 @@ jobs:
         docs/_build/
         docs/generated/
         .tox/sample_data/
-      cache-key: docs-${{ github.run_id }}
+      cache-key: ${{ needs.compute-cache-key.outputs.cache-key }}
```
Comment (nabobalis) [inline .github/workflows/ci.yml:74]: There is another:
```
          cache-key: docs-${{ github.run_id }}
```
under the gallery build, is that fine to stay or does that need the same hash as this?

File: .gitignore [hunk 2]
```diff
@@ -5,6 +5,7 @@ __pycache__/
 *.py[cod]
 *$py.class
 tmp/
+/fakedirectory/
```
Comment (nabobalis) [inline .gitignore:8]:
```suggestion
/properlyfakedirectory/
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -83,7 +95,7 @@ jobs:
             docs/_build/
             docs/generated/
             .tox/sample_data/
-          cache-key: docs-${{ github.run_id }}
+          cache-key: ${{ needs.compute-cache-key.outputs.cache-key }}
```
Comment (nabobalis) [inline .github/workflows/ci.yml:98]: No this didnt work.

**Comment by nabobalis** [comment]
I am going to make random changes for fun.

**Comment by alasdairwilson** [comment]
So yeah my attempt obv didnt work but run.id is unique per run so it would never find a cache?

I don't know why what I did didn't work.

Might be worth testing with just a static cache key first tbf.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -59,7 +71,7 @@ jobs:
         docs/_build/
         docs/generated/
         .tox/sample_data/
-      cache-key: docs-${{ github.run_id }}
+      cache-key: ${{ github.sha }}
```
Comment (nabobalis) [inline .github/workflows/ci.yml:74]: Testing if just the sha makes this work. 

Then we need to work out a more complex hashing scheme.

**Comment by Cadair** [comment]
The cache restore seems to have worked (I think it was before?) but the issue is that sphinx thinks all the source files have changed: https://github.com/sunpy/sunpy/actions/runs/9786453564/job/27021782660?pr=7703#step:10:696

**Comment by Cadair** [comment]
To be clear, I tried this locally and got the same result. In a clean git directory I ran `tox -e build_docs` twice and in the second run I get this:
```
building [html]: targets for 0 source files that are out of date
updating environment: 0 added, 414 changed, 0 removed
```

I am getting the feeling we have looked into this before. @ayshih do you remember anything about this?

**Comment by Cadair** [comment]
Ah maybe what we did before was this: https://github.com/sunpy/sunpy/issues/7268 but that doesn't seem to have resolved the fact the cache is bork.

**Comment by Cadair** [comment]
Actually I think run_id should work for us. The idea of this cache is to make the gallery build not have to rebuild the whole doc tree. According to [the docs](https://docs.github.com/en/actions/learn-github-actions/contexts) run ID is:

> A unique number for each workflow run within a repository. This number does not change if you re-run the workflow run.

That sounds right?

**Comment by alasdairwilson** [comment]
> Actually I think run_id should work for us. The idea of this cache is to make the gallery build not have to rebuild the whole doc tree. According to [the docs](https://docs.github.com/en/actions/learn-github-actions/contexts) run ID is:
> 
> > A unique number for each workflow run within a repository. This number does not change if you re-run the workflow run.
> 
> That sounds right?

You are, of course, correct.

I had interpreted it as a per job id rather than an id for the entire workflow:

I guess that means the original cache key worked but yet the cache...does not.

Closing this I guess.

**Pull request closed without merging**